### ISSUE TITLE
[Serious] Accordion/Toggle: Accordion (toggle) is missing appropriate roles and/or attributes

### ISSUE THREAD

**Issue by ntallman** [APTrust/registry#78]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87915 |
| **Checkpoint** | 4.1 Compatible › Custom Controls (4.1.2.b) |
| **Impact** | Serious |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 03 Objects page |
| **Test Case Name** | AP Trust Registry (Demo) |
| **URL** | <https://demo.aptrust.org/objects?state=A> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Product** | Website |
| **Date Reported** | 05/17/26 at 12:11 pm |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | kim.casey@ablr360.com |

## Description

The element appears and functions like an accordion (show/hide toggles) but is missing required ARIA role(s) and/or attribute(s).
Select an Object from the table, on Object Detail page, in Active Files section table, each row can be expanded to reveal more content, but these buttons are not keyboard accessible and do not convey their purpose or current state.

## Recommendation

Rule : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

How To Fix: 
Fix this issue by doing ALL of the following:
1. Put the title of each accordion/toggle header in a <button> (or ARIA equivalent) element.
2. Wrap the button in a heading (or ARIA equivalent) element.
3. When the hide/show panel associated with an accordion/toggle header is visible, set the aria-expanded attribute on the button element to true. When the panel is not visible, set aria-expanded to false.

<h3><button aria-expanded="true">When will I receive my refund?</button></h3>
<div class="faq-expanded">You should receive your refund within 5 business days....</div>

See the full ARIA accordion design pattern including optional usability enhancements and expected keyboard interactions at: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/

Reference: 
WAI-ARIA Authoring Practices: https://www.w3.org/WAI/ARIA/apg/patterns/accordion/

Background: 
Every user interface control must have a role along with any applicable states and properties so that screen reader users know how to interact with the control. Native HTML elements - such as <button>, <a>, <input>, <select> - already have a role, and their necessary states and properties - such as the checked/unchecked state of a checkbox - are automatically conveyed so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant role(s) and any applicable states and properties using ARIA as well as expected keyboard interactions.

## Source Code

```
<td class="clickable" data-toggle-table-row="file-452631" data-toggle-initialized="true">[+]</td>
```

## Screenshots

![03_keyboard](https://raw.githubusercontent.com/APTrust/registry/master/docs/accessibility-audit/screenshots/87915/03_keyboard.png)
